lib/spamcheck/spamcheck.go

```diff
@@ -4,10 +4,11 @@ import "fmt"
 
 // Request is a request to check a message for spam.
 type Request struct {
-	Msg      string   `json:"msg"`       // message to check
-	UserID   string   `json:"user_id"`   // user id
-	UserName string   `json:"user_name"` // user name
-	Meta     MetaData `json:"meta"`      // meta-info, provided by the client
+	Msg       string   `json:"msg"`        // message to check
+	UserID    string   `json:"user_id"`    // user id
+	UserName  string   `json:"user_name"`  // user name
+	Meta      MetaData `json:"meta"`       // meta-info, provided by the client
+	CheckOnly bool     `json:"check_only"` // if true, only check the message, do not update the storage state
```

Comment:
This comment specifies the media which would or wouldn't be updated but doesn't specify what is updated. Would be great to add more details on what could be updated otherwise, as now by reading the diff it was not clear why whitelist was updated unexpectedly.